static/app/views/profiling/profileDetails/hooks/useFuseSearch.ts

```diff
@@ -0,0 +1,27 @@
+import {useCallback, useMemo} from 'react';
+import Fuse from 'fuse.js';
+
+export function useFuseSearch<T extends Record<string, unknown>>(
+  data: T[],
+  options: Fuse.IFuseOptions<T>
+) {
+  const searchIndex = useMemo(() => {
+    return new Fuse(data, options);
+    // purposely ignoring options as it will cause the effect to infinitely run
```

Comment:
What effect is it causing to infinitely run?

Reply:
if you do shit like this in react;

```ts
const inlineObjectWithoutMemoization = { /**/ }

useEffect(() => {}, [inlineObjectWithoutMemoization])
```

it will run the effect on every render since technically on every render a new inline object is created, even if its values do not change.

in this case, `options` isn't memoized and results in this behaviour. 

the alternative solution to this is having the caller memoize options before passing it or making this hook check the primitive types within the object only.

i think it's a bit of an over optimization at that point since your search index ideally would never really change unless you're feeding it more data.

Reply:
That makes sense, let's keep it like this then.